docs/KettleTestingFramework.md

```diff
@@ -0,0 +1,442 @@
+
+<table>
+    <thead>
+        <tr>
+            <th colspan="3">Options accepted by the <code>kettle.test.assertErrorResponse</code> helper function</th>
+        </tr>
+        <tr>
+            <th>Option</th>
+            <th>Type</th>
+            <th>Description</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td><code>string</code></td>
+            <td><code>String</code></td>
+            <td>The returned request body from the HTTP request</td>
+        </tr>
+        <tr>
+            <td><code>request</code></td>
+            <td><code>Component</code></td>
+            <td>The <code>kettle.test.request.http</code> component which fired the request whose response is being tested</td>
+        </tr>
+        <tr>
+            <td><code>statusCode</code></td>
+            <td><code>Number</code> (default: 500)</td>
+            <td>The expected HTTP status code in ther response</td>
+        </tr>
+        <tr>
+            <td><code>errorTexts</code></td>
+            <td><code>String/Array of String</code></td>
+            <td>A single <code>String</code> or array of <code>String</code>s which must appear at some index within the <code>message</code> field of the returned JSON response payload</td>
+        </tr>
+    </tbody>
+</table>
+
+### Using cookies with an HTTP testing request
+
+A framework grade, `kettle.test.request.httpCookie`, derived from `kettle.test.request.http`, will cooperate with a component of type `kettle.test.cookieJar` which must be
+configured higher in the component tree in order to store and parse cookies. The `kettle.test.cookieJar` is automatically configured as a child of the overall `fluid.test.testCaseHolder`, but
+unless the `kettle.test.request.httpCookie` grade is used for the testing request, any returned cookies will be ignored. The `fluid.test.testCaseHolder` accepts an option, <code>secret</code>, which is
+broadcast both to the server and to the cookieJar (using Infusion's [distributeOptions](http://docs.fluidproject.org/infusion/development/IoCSS.html) directive) in order to enable them to cooperate on transmitting signed cookies.
+Consult the framework tests at [tests/shared/SessionTestDefs.js](../tests/shared/SessionTestDefs.js) for examples of how to write a sequence of HTTP fixtures enrolled in a session by means of returned cookies, both signed and unsigned.
+
+These tests are also a good example of configuring custom [middleware](#working-with-middleware) into the middle of a request's middleware chain. These tests include a middleware grade named `kettle.tests.middleware.validateSession` which
+will reject requests without a particular piece of populated session data, before processing reaches the request's `requestHandler`.
+
+### Issuing WebSockets testing fixtures with `kettle.test.request.ws`
```

Comment:
I'm thinking that it would be good to add a note on the `connect` invoker of `kettle.test.request.ws`. For example: https://github.com/simonbates/nexus/blob/master/tests/BindModelTests.js#L93-L99

We document `onConnect` but not `connect`.